runtime/lua/vim/iter.lua

```diff
@@ -179,6 +194,59 @@ function ListIter.filter(self, f)
   return self
 end
 
+--- Flattens an |list-iterator| pipeline, un-nesting nested values.
+--- The flattened list will exclude nil, and any values with non-numeric keys
```

Comment:
If this only works on list-like tables then the docstring should mention that (use other list-like only functions in this file as a reference).

Reply:
yes, I used the same language from slice: https://github.com/rudiejd/neovim/blob/f727dd00166e60534bde9001795d849fd40f5ac0/runtime/lua/vim/iter.lua#L855 

namely the `|list-iterator|` part